src/Streamly/Internal/Data/Stream/Async/Channel/Append.hs

```diff
@@ -274,17 +1011,32 @@ getLifoSVar mrun cfg = do
             , readOutputQ      = readOutput sv
             , postProcess      = postProc sv
             , workerThreads    = running
-            , workLoop         = wloop q sv
-            , enqueue          = enqueueLIFO sv q
-            , eagerDispatch    = when eager $ void $ dispatchWorker 0 sv
-            , isWorkDone       = workDone sv
-            , isQueueDone      = workDone sv
+
+            , workLoop =
+                if inOrder
+                then workLoopAhead aheadQ outH sv
+                else wloop q sv
+            , enqueue =
+                \inner ->
+                    if inOrder
+                    then enqueueAhead sv aheadQ
+                    else enqueueLIFO sv q inner
+            , eagerDispatch = when eager $ void $ dispatchWorker 0 sv
+            , isWorkDone =
+                if inOrder
+                then isWorkDoneAhead sv aheadQ outH
+                else workDone sv
+            , isQueueDone =
+                if inOrder
+                then isQueueDoneAhead sv aheadQ
+                else workDone sv
```

Comment:
I've only reviewed this part properly. I assume most of the code is copied from the other module. If not, let know.